components/tooltip/test/tooltip-help.visual-diff.js

```diff
@@ -0,0 +1,98 @@
+/*global forceFocusVisible */
+import { getRect, show } from './tooltip-helper.js';
+import puppeteer from 'puppeteer';
+import VisualDiff from '@brightspace-ui/visual-diff';
+
+describe('d2l-tooltip-help', () => {
+
+	const visualDiff = new VisualDiff('tooltip-help', import.meta.url);
+	const tooltipHelpSelector = '#help-tooltip-basic';
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await visualDiff.createPage(browser, { viewport: { width: 400, height: 400 } });
+		await page.goto(`${visualDiff.getBaseUrl()}/components/tooltip/test/tooltip-help.visual-diff.html`, { waitUntil: ['networkidle0', 'load'] });
+		await page.bringToFront();
+	});
+
+	afterEach(async() => {
+		await page.reload();
+	});
+
+	after(async() => await browser.close());
+
+	[
+		{ case: 'help tooltip hidden' },
+		{ case: 'help tooltip hovered and focused', hover: true, focus: true },
+		{ case: 'help tooltip hovered', hover: true },
+		{ case: 'help tooltip focused', focus: true },
+		{ case: 'help tooltip clicked', click: true }
+	].forEach((testCase) => {
+
+		it(testCase.case, async function() {
+			const openEvent = page.$eval(tooltipHelpSelector, async(elem) => {
+				return new Promise((resolve) => {
+					const tooltip = elem.shadowRoot.querySelector('d2l-tooltip');
```

Comment:
It looks like the `d2l-tooltip-show` event is composed, so it should be possible to avoid querying the internal shadowDOM to locate the tooltip.  Ideally tests should treat the `d2l-tooltip-help` as a "black box" from a testing perspective.  So I _think_ it should be possible to just have `elem.addEventListener('d2l-tooltip-show', resolve)`;

Reply:
Yup, you're right, that works and makes sense that it would work. Previously I had that but my tests were failing so I think as part of my debugging and experimenting, I ended up just directly setting it and left it at that once I had those tests passing.

Reply:
Brought it back though, since I agree that it feels weird to dig into the internals of the component that we're testing like that.